modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -13,14 +13,246 @@
  */
 
 (function () {
+	const HEADING_BOTH = 'Both';
+	const HEADING_COL = 'Column';
+	const HEADING_NONE = 'None';
+	const HEADING_ROW = 'Row';
+
 	const pluginName = 'toolbarbuttons';
 
 	if (CKEDITOR.plugins.get(pluginName)) {
 		return;
 	}
 
+	/**
+	 * Creates a new CKEDITOR.dom.element using the passed tag name.
+	 *
+	 * @instance
+	 * @memberof CKEDITOR.Table
+	 * @protected
+	 * @method _createElement
+	 * @param {String} name The tag name from which an element should be created
+	 * @param {Object} editor The CKEditor instance.
+	 * @return {CKEDITOR.dom.element} Instance of CKEDITOR DOM element class
+	 */
+	function _createElement(name, editor) {
+		return new CKEDITOR.dom.element(name, editor.document);
+	}
+
+	/**
+	 * Returns which heading style is set for the given table.
+	 *
+	 * @instance
+	 * @memberof CKEDITOR.Table
+	 * @method getHeading
+	 * @param {CKEDITOR.dom.element} table The table to gather the heading from. If null, it will be retrieved from the current selection.
+	 * @param {Object} editor The CKEditor instance.
+	 * @return {String} The heading of the table. Expected values are `CKEDITOR.Table.NONE`, `CKEDITOR.Table.ROW`, `CKEDITOR.Table.COL` and `CKEDITOR.Table.BOTH`.
+	 */
+	function getHeading(table, editor) {
```

Comment:
It is quite unusual to have methods outside of the plugin, in a closure. I'm referring to `getHeading`, `getFromSelection` and `getFromSelection` here. This pattern is usually used to implement a truly private property of an object, something that we are not doing here. These should probably be properties on the plugin object, accessed with `this._getHeading`,  `this._getFromSelection` and `this._getFromSelection`

As methods of a plugin, we don't need to have jsdoc on them. I mean we could, but we are not doing this anywhere else in other plugins.

Reply:
The idea here was to isolate from side effects since I'm not a fan of allowing the possibility of mixing the scope of the function with the plugin stuff but it could be a good idea to prevent passing editor argument every call. However, we could use wrong editor instances and I'm fear this from last experiences

Reply:
I guess it makes sense, to highlight them as utility functions. Thinking about it now, I am not sure if we need anything in the plugin object, besides the `init` implementation :thinking: In other plugins we typically add everything to the object. This actually looks to be like a better pattern :+1: